collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java

```diff
@@ -151,8 +155,10 @@ private void initializeMapFrameworks() {
 
     private void importExistingProjectIfNeeded() {
         if (!appStateProvider.isFreshInstall() && !metaSettings.getBoolean(MetaKeys.EXISTING_PROJECT_IMPORTED)) {
-            projectImporter.importExistingProject();
+            Project.Saved existingProject = projectImporter.importExistingProject();
+            currentProjectProvider.setCurrentProject(existingProject.getUuid());
```

Comment:
What's the benefit of moving this out?

Reply:
You mean the `CurrentProjectProvider` out of `ProjectImporter`?  I was just trying to have the `...Importer` focus on the setup/creation of the project. I was finding it a bit overwhelming having one component that created the project and switched to it. That might get simpler once we've done some rework for background jobs I guess. It could also get worse though 😕